### ISSUE TITLE
[NBug] Index was out of range. Must be non-negative and less than ...

### ISSUE THREAD

**Issue by Futronix** [gitextensions/gitextensions#8609]
:warning: The sections below must be filled in and this text must be removed or the issue will be closed.

## Error Details
```
System.ArgumentOutOfRangeException: Index was out of range. Must be non-negative and less than the size of the collection.
Parameter name: index
   at System.Collections.ArrayList.get_Item(Int32 index)
   at System.Windows.Forms.DataGridViewRowCollection.SharedRow(Int32 rowIndex)
   at System.Windows.Forms.DataGridViewRowCollection.get_Item(Int32 index)
   at GitUI.RevisionGridControl.OnGridViewCellMouseDown(Object sender, DataGridViewCellMouseEventArgs e)
   at System.Windows.Forms.DataGridView.OnCellMouseDown(DataGridViewCellMouseEventArgs e)
   at System.Windows.Forms.DataGridView.OnMouseDown(MouseEventArgs e)
   at GitUI.UserControls.RevisionGrid.RevisionDataGridView.OnMouseDown(MouseEventArgs e)
   at System.Windows.Forms.Control.WmMouseDown(Message& m, MouseButtons button, Int32 clicks)
   at System.Windows.Forms.Control.WndProc(Message& m)
   at System.Windows.Forms.DataGridView.WndProc(Message& m)
   at System.Windows.Forms.Control.ControlNativeWindow.OnMessage(Message& m)
   at System.Windows.Forms.Control.ControlNativeWindow.WndProc(Message& m)
   at System.Windows.Forms.NativeWindow.Callback(IntPtr hWnd, Int32 msg, IntPtr wparam, IntPtr lparam)
```

## Additional information
Git extension became very slow over the last week up to a point that it is now unsable...

## Environment
- Git Extensions 3.4.3.9999
- Build d4b0f48bbf3e39a71f5d7ff5231be5678d4aa0f1
- Git 2.28.0.windows.1
- Microsoft Windows NT 10.0.19041.0
- .NET Framework 4.8.4250.0
- DPI 120dpi (125% scaling)

**Comment by gerhardol** [comment]
Duplicate of #7937 

Fixed in 3.4

**Comment by gerhardol** [comment]
Duplicate of #7374
Corrected in #8534, please try the 3.5 RC1
(There was a correction for #7374 in #7921 that is in 3.4)
